cli/src/pcluster/resources/login_node/user_data.sh

```diff
@@ -114,7 +114,13 @@ write_files:
         # wait logs flush before signaling the failure
         sleep 10
         # TODO: add possibility to override this behavior and keep the instance for debugging
-        shutdown -h now
+        # Notify the AutoScalingGroup to proceed with adding the instance for service
+        IMDS_TOKEN=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 300")
+        INSTANCE_ID=$(curl -H "X-aws-ec2-metadata-token: $IMDS_TOKEN" -v http://169.254.169.254/latest/meta-data/instance-id)
```

Comment:
[Nitpick] Since what we need is the INSTANCE_ID we could invoke IMDS only once at the beginning of the script and reuse the INSTANCE_ID value in the whole script.

We can improve this in a following PR.